src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -61,15 +61,14 @@ struct KeyValueViewModel: Comparable {
         - Parameter operation: the operation which this paramter exists.
      */
     init(from param: ParameterType, with operation: Operation) {
-        let name = param.serializedName ?? param.name
-
         if let constantSchema = param.schema as? ConstantSchema {
-            self.init(param: param, constantSchema: constantSchema, name: name)
-        } else if let signatureParameter = operation.signatureParameter(for: name) {
+            self.init(param: param, constantSchema: constantSchema)
+        } else if let signatureParameter = operation.signatureParameter(for: param.name) {
             self.init(signatureParameter: signatureParameter)
         } else if let groupedBy = param.groupedBy?.name {
-            self.init(key: name, value: "\(groupedBy).\(name)")
+            self.init(key: param.name, value: "\(groupedBy).\(param.name)")
         } else if param.implementation == .client {
+            let name = param.name.lowercasedFirst
```

Comment:
It seems a little random where we are applying `lowercasedFirst`. Can we make it so that the getter for name is always lowercasedFirst?

Reply:
I have tried that approach (that the getter for name is always lowercasedFirst) first, but it breaks many other cases where we need to name to be upper case as First letter.  E.g. Model name, Options name, Enum Names, etc.
So that is why call `lowercasedFirst` at the places we needed.